src/analysis.py

```diff
@@ -179,10 +196,10 @@ def train_models(train_df, models, param_grid=None, output_dir=""):
     X_train = train_df.drop(columns=['align'])
     y_train = train_df['align']
 
-    numeric_features = ['appearances']
-    categorical_features = ['name', 'id', 'eye', 'hair', 'publisher', 'sex']
-    ordinal_features = ['year']
-    drop_features = ['gsm', 'first_appearance']
+    numeric_features = ['appearances', 'year', 'name_len', 'appear_per_yr']
+    categorical_features = ['id', 'eye', 'hair', 'publisher', 'sex']
+    drop_features = ['name', 'gsm', 'first_appearance', 'first_name', 'last_name']
+    binary_features = ['is_common', 'last_name']
```

Comment:
Maybe replace 'last_name' with 'has_last_name'?

Reply:
Thank you for catching - I will update tomorrow morning, test, and push.